V2.13 (#697)
run-ai/docs: docs/home/changelog/hotfixes-2-13.md

---
title: Changelog Version 2.13
summary: This article lists the fixed and known issues in the patch versions as well as additional new features that were added in each patch version.
authors:
    - Jason Novich
date: 2024-Jan-22
---

The following is a list of the known and fixed issues for Run:ai V2.13.

## Version 2.13.43 - February 15, 2024

| Internal ID | Description  |
| ---------------------------- | ---- |
| RUN-14946 | Fixed an issue where Dashboards are displaying the hidden Grafana path. |

## Version 2.13.37

| Internal ID | Description  |
| ---------------------------- | ---- |
| RUN-13300 | Fixed an issue where projects will appear with a status of empty while waiting for the project controller to update its status. This was caused because the cluster-sync works faster than the project controller. |

## Version 2.13.35 - December 19, 2023

### Release content

* <!-- RUN-14441 Add ability to set node affinity for Prometheus -->Added the ability to set node affinity for Prometheus.

### Fixed issues

| Internal ID | Description  |
| ---------------------------- | ---- |

> | RUN-14472 | Fixed an issue where template updates were not being applieed to the workload. |

Reviewer: Applieed

| RUN-14434 | Fixed an issue where `runai_allocated_gpu_count_per_gpu` was multipled by seven. |

> | RUN-13956 | Fixed an issue where changing an exisitng template created a `Promise error` on existing job templates. |

Reviewer: ....when changing an existing template in the Jobs screen...

Reviewer 2: I think it should be "multiplied"

> | RUN-13825 | Fixed an issue where GPU sharing configmaps are leaked. |

Reviewer: Very frightening and not relevant, plase remove this bug

Reviewer 2: I don't know what "GPU sharing" means in this context. If swap - not interesting. If fractions, we can phrase that more gently as "when deleting a job that is allocated a fraction of a GPU, an associated configmap is not deleted"

Author: @omerbenedictrunai 
```suggestion
| RUN-13825 | Fixed an issue when deleting a job that is allocated a fraction of a GPU, an associated configmap is not deleted. |
```

| ---------------------------- | ---- |
| RUN-12563 | Fixed an issue where users are unable to login after upgrading the control plane from 2.9.16 to 2.13.16. To correct the issue, secrets need to be upgraded manually in keycloak. |

## Version 2.13.20 - September 28, 2023

### Release content

* <!-- RUN-12537 hide credentials from the UI when cluster version is not compatible -->Added the prevention of selecting tenant or department scopes for credentials, and the prevention of selecting s3, PVC, and Git data sources if the cluster version does not support these.
* <!--RUN-12349	Enable quota management by default -->Quota management is now enabled by default.

| Internal ID | Description  |
| ---------------------------- | ---- |

> | RUN-12923 | Fixed an issue in upgrading due to a misconfigured image for airgapped systems in 2.13.19. The helm chart contained an error, and the image is not used even though it is packaged as part of the tar. |

Reviewer 2: image --> docker image

Author: ```suggestion
| RUN-12923 | Fixed an issue in upgrading due to a misconfigured Docker image for airgapped systems in 2.13.19. The helm chart contained an error, and the image is not used even though it is packaged as part of the tar. |
```

| RUN-12928, </br>RUN-12968 | Fixed an issue in upgrading Prometheus due to a misconfigured image for airgapped systems in 2.13.19. The helm chart contained an error, and the image is not used even though it is packaged as part of the tar. |
| RUN-12751 | Fixed an issue when upgrading from 2.9 to 2.13 results with a missing engine-config file. |
| RUN-12717 | Fixed an issue where the user that is logged in as researcher manager can't see the clusters. |

> | RUN-12642 | Fixed an issue where Assets-sync could not restart due to failing to get token from control plane. |

Reviewer 2: See comment @jasonnovichRunAI

> | RUN-12642 | Fixed an issue where Assets-sync could not restart due to failing to get token from control plane. |

Author: ```suggestion
| RUN-12642 | Fixed an issue where assets-sync could not restart due to failing to get token from control plane. |
```

> | RUN-12191 | Fixed an issue where there was a Time out while waiting for the `runai_allocated_gpu_count_per_project` metric to return values. |

Reviewer: timeout - one word no capital 'T'

| RUN-10474 | Fixed an issue where the `runai-conatiner-toolkit-exporter` DaemonSet fails to start. |

## Version 2.13.19 - September 27, 2023

### Release content

* <!--RUN-12508  Identify and present Kubeflow Notebooks in Jobs table -->Added the ability to identify Kubeflow notebooks and display them in the Jobs table.
* <!-- RUN-12507 Support scheduling of Kubeflow workloads -->Added the ability to schedule Kubelow workloads.
* <!--RUN-12420 Show only logged-in user's jobs -->Added functionality that displays Jobs that only belong to the user that is logged in.
* <!-- RUN-12405 Refine out of the box alerts to run:ai rules --> Added and refined alerts to the state of Run:ai components, schedule latency, and warnings for out of memory on Jobs.
* <!-- RUN-11711 Pod Security Admission issues in control plane -->Added the ability to work with restricted PSA policy.

### Fixed issues

| Internal ID | Description  |
|-------------- | ------------------- |
| RUN-12650 | Fixed an issue that used an incorrect metric in analytics GPU ALLOCATION PER NODE panel. Now the correct allocation is in percentage. |

> | RUN-12602 | Fixed an issue in `runaiconfig` where `WorkloadServices` have any memory requests/limits and cpu requests/limits and is overwritten with a default. |

Reviewer 2: I don't understand the statement as written

Author: From what this bug looks like, it seems that there are settings in the `runaiconfig` in the `workloadservices` section that were overwritten with default settings. Meaning the confgig was ignored, superseded, or not working (essence of the bug). I can rephrase to:

```suggestion
| RUN-12602 | Fixed an issue in `runaiconfig` where the `WorkloadServices` spec has memory requests/limits and cpu requests/limits and gets overwritten with the system default. |
```

> | RUN-12587 | Fixed an issue where the scheduler enters an endless loop running elastic jobs. |

Reviewer 2: What is an "elastic job"?

Author: @yarongol I am going to assume that it means this https://kubernetes.io/docs/concepts/workloads/controllers/job/#elastic-indexed-jobs. Correct me if I didn't get it, but it seems that an elastic job is a form of autoscaling jobs. So maybe the scheduler has an issue with autoscaling and the above API reference?

Author: @omerbenedictrunai 
```suggestion
| RUN-12587 | Fixed an issue where elastic scheduling of jobs enters an endless loop. |
```

| RUN-12585 | Fixed an issue where the workload-controller creates a delay in running jobs. |
| RUN-12031 | Fixed an issue when upgrading from 2.9 to 2.13 where the Scheduler pod fails to upgrade due to the change of owner. |

> | RUN-11091  | Fixed an issue when `departments FF` is off you are not able to schedule non-preemable jobs. |

Reviewer: department FF is not customer facing. Use th following customer facing option:
"Enable Departments" option under Settings

Author: @omerbenedictrunai
```suggestion
| RUN-11091  | Fixed an issue where the *Departments* feature is disabled, you are not able to schedule non-preemable jobs. |
```
